CRM/Utils/Check.php

```diff
@@ -209,7 +212,7 @@ public static function checkAll($max = FALSE) {
 
     foreach ($messages as $key => $message) {
       $hush = self::checkHushSnooze($message);
-      $message->setVisible(!$hush);
+      $messages[$key]->setVisible(!$hush);
```

Comment:
Thanks for the explanation @totten and @colemanw.  I had never tried it in the context of a foreach loop.  (Or so I thought--turns out that line was added by @jdax as I was sitting next to her.  I'm just silly.)  Anyway, I won't be offended if you want to change it back to `$message->setVisible(!$hush);`